### PR TITLE
Update retry count to 10

### PR DESCRIPTION
**Pull request by carlweis**
About 10 times per month, we fail to create sharing links within the platform (e.g., unique tracking links, social sharing button links, personal plea links, etc.) and the page crashes.

We have about 1.7mil of these links within our database, and the generation process is random. While there should be billions of possible combinations for the shortened URL's we're randomly generating, we occasionally run into naming collisions.

Unfortunately, after trying to generate a unique URL 5 times in a row, the code gives up and likely displays a Whoops! page to the user in question.

Retrying this logic isn't resource-intensive, so let's bump the amount of possible retries to significantly reduce the possibility of this edge case in the future.

Marking as Medium priority, since even though this happens rarely, it can still crash the campaign page completely and lead to a school losing out on a $1000 gift.

For Engineering:

We need to update this line within the shortener Ruby project to increase the retry_count to something like 10

We'll then need to update the Gemfile within the givecampus project to reflect our changes.

Example Server Log

https://app.asana.com/0/1172077698943204/1199397100687775

### CODE REVIEW

**Review by jacob-comer — APPROVED** [review]

File: app/models/shortener/shortened_url.rb [hunk 1]
```diff
@@ -66,7 +66,7 @@ def self.generate(orig_url, user=nil, project_id)
       self.unique_key = generate_unique_key
       super()
     rescue ActiveRecord::RecordNotUnique, ActiveRecord::StatementInvalid => err
-      if (count +=1) < 5
+      if (count +=1) < 10
```
Comment (grubmatt) [inline app/models/shortener/shortened_url.rb:69]: generate_unique_key works by picking a random character from a list of 36 options 6 times in a row.

So the chance of any single key being picked is (1/36) * (1/36) * (1/36) * (1/36) * (1/36) * (1/36) this is equivalent to 4.59 * 10^-10 or 0.000000000459.

Similarly, there are 2,176,782,336 possible keys this field could be or 1 / 2,176,782,336 chances that a key is duplicated (which is the same value as above but approached from a different way). The odds of this happening 5 times in a row is (4.59 * 10^-10 )^5 which is equivalent to 2.04*10^47. 

Furthermore, looking at the logs: https://my.papertrailapp.com/events?q=retrying%20with%20different%20unique%20key&selected=1270214948108206082 these errors are always appearing in pairs of 4.

The odds of this happening once are 2.04*10^47, the odds of this happening twice is 2.04*10^47  * 2.04*10^47 or 4.16*10^94. 

For context, it is estimated that there are 10^80 atoms in the universe. So there is a better chance of selecting a random atom from the universe than there is that this same error could pop up two times. It has happened over 20 times in the past month.

Based on these odds I suspect we are falling into this rescue because of `ActiveRecord::StatementInvalid`.

I'm not sure what would be causing this so could we start logging err to get to the bottom of it?

---

TL;DR I did some probability for fun, it's pretty much impossible that the error we're running into is due to: `ActiveRecord::RecordNotUnique`.

Could we log the err to get a better idea as to why this is popping up:

```
rescue ActiveRecord::RecordNotUnique, ActiveRecord::StatementInvalid => err
  logger.info("Shortener gem attempted to generate a unique key and raised the following err: #{err}")
  if (count +=1) < 10
    logger.info("retrying with different unique key")
    retry
  else
    logger.info("too many retries, giving up")
    raise
  end
end
```

Comment (carlweis) [reply app/models/shortener/shortened_url.rb:69]: @grubmatt Yeah...that's a good idea. Could you add the logging?
